install-deps.sh

```diff
@@ -100,6 +100,62 @@ EOF
     fi
 }
 
+function ensure_min_npm_version {
+  local install_npm_pkg_cmd=$1
+
+  if ! type npm > /dev/null 2>&1; then
```

Comment:
is `type` the standard way of checking for a command's existence in the PATH? I would presume `which` or `whereis` would make more sense. Especially because one can do funny things that `type` will not catch, e.g.:

> $ alias npm=echo
> $ type npm
> npm is aliased to `echo'
> $ echo $?
> 0

Reply:
I don't know. I saw `type` being used in other part of `install-deps.sh` script and did the same. 
@smithfarm do you how to answer to @jecluis question?

Reply:
we started migrating from `which` to `type` after it became clear that "which" is not guaranteed to be installed on all systems - especially minimalistic ones. Although it's possible someone might have an `npm` alias without npm actually installed, but to me that's a corner case that the person will just have to deal with.

I had never heard of `whereis`, but on my system `whereis foobarblatz` returns 0 status code.

Reply:
I'm satisfied. ty.